src/main/Port.Adapter/UI/Views/Blazor.Common/TreeView.razor

```diff
@@ -260,46 +260,69 @@
     {
         if (
             this.selectedNeuron != null && string.IsNullOrEmpty(this.selectedNeuron.Neuron.ExternalReferenceUrl) &&
-            this.selectedNeuron.IsExpandTimerEnabled() && 
+            this.selectedNeuron.IsExpansionTimerEnabled() &&
             (
                 this.selectedNeuron.ExpansionState == ExpansionState.Collapsed ||
                 !this.Children.Any(c => c.ExpansionState == ExpansionState.Expanded)
             )
         )
         {
-            this.selectedNeuron.RestartExpandTimer();
+            this.selectedNeuron.RestartExpansionTimer();
 
-            // if children contains selected neuron
+            switch (this.selectedNeuron.CurrentExpansionType)
+            {
+                case ExpansionType.PostsynapticUntilExternalReferences:
+                    await HandleExpansion(
+                        Library.Common.RelativeType.Presynaptic, 
+                        Library.Common.RelativeType.Postsynaptic,
+                        n => string.IsNullOrEmpty(n.ExternalReferenceUrl));
+                    break;
+                case ExpansionType.FarthestPresynaptic:
+                    await HandleExpansion(
+                        Library.Common.RelativeType.Postsynaptic, 
+                        Library.Common.RelativeType.Presynaptic,
+                        _ => true);
```

Comment:
Since the last parameter is optional, we no longer need to pass it in. Please remove the parameter "_ => true".